src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -172,13 +172,17 @@ public boolean isPinned()
     @Override
     public RestAction<Void> clearReactions(@Nonnull String unicode)
     {
+        if (!isFromType(ChannelType.TEXT))
+            throw new IllegalStateException("Cannot clear reactions from a message in a Group or PrivateChannel.");
         return getTextChannel().clearReactionsById(getId(), unicode);
     }
 
     @Nonnull
     @Override
     public RestAction<Void> clearReactions(@Nonnull Emote emote)
     {
+        if (!isFromType(ChannelType.TEXT))
+            throw new IllegalStateException("Cannot clear reactions from a message in a Group or PrivateChannel.");
```

Comment:
You should probably remove the group from here since bots can't be in groups

Reply:
I am fully aware of this but since this was [the original output](https://github.com/DV8FromTheWorld/JDA/blob/fb0c6281a3eec4c07fc33ca1699d417077ada61b/src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java#L166-L167) and the throws section saying ["if was not sent in a TextChannel"](https://github.com/DV8FromTheWorld/JDA/blob/fb0c6281a3eec4c07fc33ca1699d417077ada61b/src/main/java/net/dv8tion/jda/api/entities/Message.java#L1190-L1192), I decided to stay with this.

However in my JavaDoc additions, I explicitly talk about private messages since it's easier to understand that than telling it has to come from a TextChannel imo.

TL;DR: I would wait for Minn's response since this is likely to be unwanted.